### ISSUE TITLE
App does not run when adding .framework to iOS plugin

### ISSUE THREAD

**Issue by YazeedAlKhalaf** [flutter/flutter#85951]
I am currently making a flutter plugin. Now working on the iOS part. My plugin depends on a .framework that is objective-c code. The plugin code is in Swift.

I was able to run the app successfully if the plugin code was in the AppDelegate.swift and the import of the objective-c framework is in the Runner bridging header. But when I copy the code to a plugin and try to run the code it does not work.

I get an error saying that:

```
Include of non-modular header inside framework module 'myplugin.myplugin_Bridging_Hearer'
```

I have added the framework to podspec using these three lines:

```
Will be added from desktop
```

XCode does not complain before build. The swift code does not have any errors. If I remove the lines from podspec I get the framework does not exist instead of non-module error.

**Comment by TahaTesser** [comment]
Hi @YazeedAlKhalaf 
This looks similar to https://github.com/flutter/flutter/issues/16049. where your project is in swift and other is in objective-c

**Comment by YazeedAlKhalaf** [comment]
> Hi @YazeedAlKhalaf 
> 
> This looks similar to https://github.com/flutter/flutter/issues/16049. where your project is in swift and other is in objective-c
> 
> Thanks for filing the issue

Yes similar but not the same. Exactly what you said. I tried some workarounds proposed in there but none worked.

**Comment by pr-Mais** [comment]
Facing same issue here. 
The example app runs when commenting out `use_frameworks!`, not sure about this behavior.

**Comment by pr-Mais** [comment]
We found a workaround in this package [flutter_txugcupload](https://github.com/leanflutter/flutter_txugcupload/blob/master/README.md), applied it to our case [here](https://github.com/nyartech/hyperpay/blob/main/example/ios/Podfile) and it worked well, the app now compiles in debug.
Will still need to test it further.

**Comment by gyuri-fluttech** [comment]
I am experiencing the same problem but the solution provided by @pr-Mais solved it. Thanks.

**Comment by gyuri-fluttech** [comment]
@pr-Mais , the problem reappears when I want to create a build by `flutter build ios`. Did it work for you ?

**Comment by pr-Mais** [comment]
Yes. The code above in Podfile still works for me in my plugin.

**Comment by gyuri-fluttech** [comment]
hmm, when I run the app on my iOS device, it is working, but when I want to create a build, it gives me this: 


```
     remark: Incremental compilation has been disabled: it is not compatible with whole module optimization
    <module-includes>:1:9: note: in file included from <module-includes>:1:
    #import "azure_cognitive_services-umbrella.h"
            ^
    /Users/gyuri/example_flutter/ios/Pods/Headers/Public/azure_cognitive_services/azure_cognitive_services-umbrella.h:13:9: note: in file included from
    /Users/gyuri/example_flutter/ios/Pods/Headers/Public/azure_cognitive_services/azure_cognitive_services-umbrella.h:13:
    #import "AzureCognitiveServicesPlugin.h"
            ^
    /Users/gyuri/example_flutter/ios/Pods/Headers/Public/azure_cognitive_services/AzureCognitiveServicesPlugin.h:2:9: error: 'MicrosoftCognitiveServicesSpeech/SPXSpeechAPI.h' file not found
    #import <MicrosoftCognitiveServicesSpeech/SPXSpeechAPI.h>
            ^
    <unknown>:0: error: could not build Objective-C module 'azure_cognitive_services'
    Command CompileSwiftSources failed with a nonzero exit code
    remark: Incremental compilation has been disabled: it is not compatible with whole module optimization
    Command CompileSwiftSources failed with a nonzero exit code
    remark: Incremental compilation has been disabled: it is not compatible with whole module optimization
```

Basicall, the same issue I had before adding your snippet in `Podfile`

Am I missing something ?
